### ISSUE TITLE
AttributeError: 'float' object has no attribute 'deg2rad'

### ISSUE THREAD

**Issue by scottkehler** [Unidata/siphon#174]
I have installed siphon on both my mac and windows computers in an attempt to download upper air data from U of Wyoming. On both systems, I get the following error when trying to run the example upper air download script:

AttributeError: 'float' object has no attribute 'deg2rad'

It seems to occur when the df['direction'] object is being converted from degrees to radians:

df = pd.read_fwf(raw_data, skiprows=5, usecols=[0, 1, 2, 3, 6, 7], names=col_names) df['u_wind'], df['v_wind'] = get_wind_components(df['speed'] ,np.deg2rad(df['direction']))

I attached a screenshot of my iPython notebook for reference.
![capture](https://user-images.githubusercontent.com/19496053/34394559-93ef08ee-eb1f-11e7-889b-f31be8339b0a.JPG)

**Comment by dopplershift** [comment]
Weird, I can't reproduce here on Python 3.6 on mac. The error message would seem to indicate that `np` is a float, rather than the NumPy library. Can you see if the problem is still occurring if you restart the notebook kernel?

**Comment by scottkehler** [comment]
I tried restarting the kernel a few times and still get the error. I also tried running the script using Python 3 and got the error with that as well.

I am running the script in Canopy - not sure if that is a possible factor?

**Comment by dopplershift** [comment]
Can you:

1. Uninstall Siphon (`pip uninstall siphon`)
2. Make sure there are no files in `C:\Users\Scott\AppData\Local\Enthought\Canopy\User\lib\site-packages\siphon` (just delete the folder)
3. Reinstall siphon?

I'm kind of grasping at straws here (guessing that the install went funky), since the error indicates that the interpreter thinks `np` is a floating point value, which shouldn't be possible.

**Comment by scottkehler** [comment]
I tried uninstalling and reinstalling but that didn't work. The strange thing is that I get the same error on two different computers with different operating systems (Windows and Mac). Uninstalling and reinstalling on both systems resulted in the same error again.

**Comment by dopplershift** [comment]
Any chance this has magically solved itself on your system?

**Comment by dopplershift** [comment]
Ugh, ok. That deprecated stuff will actually go away in MetPy 0.8 (scheduled for release in mid-to-late April 2018). What version of Canopy do you have installed?

**Comment by scottkehler** [comment]
I’m using canopy version 2.1.2.3530 (64 bit).

Scott Kehler
scottdkehler@gmail.com <mailto:scottdkehler@gmail.com>

**Comment by dopplershift** [comment]
Which version of Python (3.5/2.7)?

**Comment by scottkehler** [comment]
Good news, I got it working! I didn’t actually install the newer version of canopy, all I did was install protobuf_python through the package manager and updated netcdf4 through the package manager. Siphon then started to work. I guess the issue was that the protobuf installed via pip didn’t work, so when I installed it using the package manager it fixed the issue?

Scott Kehler
scottdkehler@gmail.com <mailto:scottdkehler@gmail.com>
